backend-public/contracts/approving-namespace-controller.clar

```diff
@@ -1,71 +1,64 @@
 (define-constant namespace 0x67676767676767676767)
 
 (define-data-var contract-owner principal tx-sender)
-(define-data-var dao-treasury principal tx-sender)
+(define-data-var community-treasury principal tx-sender)
 (define-data-var approval-pubkey (buff 33) 0x00)
 (define-data-var price-in-ustx uint u4999999)
 
 (define-constant err-not-authorized (err u403))
 
 ;; register an ordered name
 ;; @event: tx-sender sends 1 stx
-;; @event: this contracts sends 1 stx
 ;; @event: community-handles burns 1 stx
 ;; @event: community-handles sends name nft to tx-sender
 (define-public (name-register (name (buff 48))
                               (approval-signature (buff 65))
                               (zonefile-hash (buff 20)))
     (let ((price (var-get price-in-ustx))
-            (salt 0x00)
-            (owner tx-sender)
-            (hash (sha256 (concat (concat (concat name 0x2e) namespace) salt))))
+          (salt 0x00)
+          (owner tx-sender)
+          (hash (sha256 (concat (concat (concat name 0x2e) namespace) salt))))
         (asserts! (secp256k1-verify hash approval-signature (var-get approval-pubkey)) err-not-authorized)
         (try! (pay-fees price))
-        (try! (stx-transfer? u1 tx-sender (as-contract tx-sender)))
-        (try! (as-contract (contract-call? .community-handles name-register namespace name salt zonefile-hash owner)))
-        (ok true)))
+        (contract-call? .community-handles name-register namespace name zonefile-hash owner)))
 
 
 (define-private (pay-fees (price uint))
     (let ((amount-ohf (/ (* price u70) u100))
-          (amount-dao (- price amount-ohf)))
+          (amount-community (- price amount-ohf)))
         (try! (stx-transfer? amount-ohf tx-sender (var-get contract-owner)))
-        (try! (stx-transfer? amount-dao tx-sender (var-get dao-treasury)))
-        (ok true)))
+        (stx-transfer? amount-community tx-sender (var-get community-treasury))))
 
 ;;
 ;; admin functions
 ;;
 (define-public (set-price (amount-in-ustx uint))
     (begin
         (try! (is-contract-owner))
-        (var-set price-in-ustx amount-in-ustx)
-        (ok true)))
+        (ok (var-set price-in-ustx amount-in-ustx))))
```

Comment:
having `(ok true)` as the last line clarifies that the function will never return `(ok false)`